traefik/templates/_helpers.tpl

```diff
@@ -219,7 +219,7 @@ Key: {{ index $.Values.hub.apimanagement.admission.customWebhookCertificate "tls
 Hash: {{ sha1sum (index $.Values.hub.apimanagement.admission.customWebhookCertificate "tls.crt") }}
 {{- else -}}
     {{- $cert := lookup "v1" "Secret" (include "traefik.namespace" .) $.Values.hub.apimanagement.admission.secretName -}}
-    {{- if $cert -}}
+    {{- if and $cert (hasKey $cert.data "tls.crt") (hasKey $cert.data "tls.key") -}}
```

Comment:
While I would tend to agree on this, I think this is very intrusive.
With this test, we know for sure this secret doesn't contain any certificate and if somehow the user provide an existing secret name by mistake for `hub.apimanagement.admission.secretName`, this code will simply erase this secret.
I think the helm chart should just fail at this point.
```suggestion
    {{- if $cert }}
        {{ if or (not (hasKey $cert.data "tls.crt")) (not (hasKey $cert.data "tls.key")) -}}
            {{- fail (printf "ERROR: secret %s/%s exists but doesn't contain any certificate data. Please remove it or change hub.apimanagement.admission.secretName." (include "traefik.namespace" .) $.Values.hub.apimanagement.admission.secretName) }}
        {{- end -}}
```